### ISSUE TITLE
Pinning of botocore dependency causes downstream version conflicts

### ISSUE THREAD

**Issue by franco-circleup** [aio-libs/aiobotocore#840]
**Describe the bug**
Cannot resolve dependencies due to aiobotocore pinning botocore dep 

**Checklist**
- [ ] I have reproduced in environment where `pip check` passes without errors
- [ ] I have provided `pip freeze` results
- [x] I have provided sample code or detailed way to reproduce
- [x] I have tried the same code in botocore to ensure this is an aiobotocore specific issue
- [x] I have tried similar code in aiohttp to ensure this is is an aiobotocore specific issue
- [x] I have checked the latest and older versions of aiobotocore/aiohttp/python to see if this is a regression / injection

**What happened**:

Using pip-tools to resolve dependency versions with the following basic requirements:

`requirements.in`:
```
boto3==1.*
moto==1.*
s3fs
```

`pip-compile requirements.in`

Result:
```
There are incompatible versions in the resolved dependencies:
  botocore>=1.11.3 (from aws-xray-sdk==2.6.0->moto==1.3.16->-r requirements.in (line 23))
  botocore<2.0a.0,>=1.12.36 (from s3transfer==0.3.3->boto3==1.16.24->-r requirements.in (line 5))
  botocore<1.17.45,>=1.17.44 (from aiobotocore==1.1.2->s3fs==0.5.1->-r requirements.in (line 37))
  botocore<1.20.0,>=1.19.24 (from boto3==1.16.24->-r requirements.in (line 5))
  botocore>=1.12.201 (from moto==1.3.16->-r requirements.in (line 23))
```

As you can see, the `s3fs` dependency on `aiobotocore`  is the issue, everything else is pretty loose. `s3fs` only requires botocore>=1.0.1, so it's not s3fs' issue. It's `aiobotocore` pinning `'botocore>=1.17.44,<1.17.45'.`

Pinning the patch version or even the minor version is not going to be a sustainable approach with boto/botocore, as they are releasing a new minor version every few months: https://pypi.org/project/boto3/#history

With the new pip resolver coming, this is not just going to affect pip-tools/pipenv/poetry users either - everyone will start seeing their installs break on a dependency conflict soon.

With bare pip:
`pip install -r requirements.in`
```
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

aiobotocore 1.1.2 requires botocore<1.17.45,>=1.17.44, but you'll have botocore 1.19.24 which is incompatible.
```

I would recommend floating the boto/botocore requirements on 1.* to maximize flexibility for end users that install this package. I think concerns about the stability of boto/botocore stability on 1.* would be minimal, if any.

It seems like a good general policy for libraries to define looser requirements (obviously avoiding breaking changes and security issues) and let end users pin them as needed.

**Environment:**
 - Python Version: 3.6
 - OS name and version: Darwin 19.6.0

**Additional context**

**Comment by franco-circleup** [comment]
I would have provided a PR, but since it's just changing a version requirement, I figured the maintainers would want to ponder how restrict the version and handle the change themselves :)

**Comment by evinism** [comment]
+1 on this one-- is there a specified reason for such a strong pin?

**Comment by andyndang** [comment]
AFAIK, AWS SDK is very backwards compatible unless it's a major version upgrade. I can't think of a reason to pin to such a specific minor version in particular

**Comment by asvetlov** [comment]
The SDK is very backward compatible in the public part. But aiobotocore is supposed to monkey-patch the SDK internals for making everything working asynchronously.
Any innocent change in SDK can break aiobotocore potentially.
That's why the pinning is important.

**Comment by evinism** [comment]
that's actually a really, really solid answer. Thanks a ton for the response

**Comment by franco-circleup** [comment]
Thanks for the response @asvetlov !

I suppose we'll have to do some further gymnastics took keep our boto requirements old enough to match aiobotocore constraints and new enough to be compatible with everything else, or find a way to drop S3fs altogether.

**Comment by andyndang** [comment]
FYI with the new pip logic, this means that nobody can use a different version of botocore. Because of this, we're dropping S3fs use in our code base since we need to make call to more crecent APIs

**Comment by steven2308** [comment]
If for some reason you also use boto3 (required by moto for instance), this is the best option I found where there's no conflict.
aiobotocore==1.1.2
botocore==1.17.44
boto3==1.14.44

Moto 1.3.16 unfortunately doesn't work straightforward with the mock decorators since the responses do not implement the raw_headers. I'm still looking for a workaround based on how aiobotore is tested.

**Comment by thehesiod** [comment]
You can just use the aioboto boto3 extra to get there right version

**Comment by thehesiod** [comment]
this is as-designed, just a matter of fact due to how aiobotocore is layered on top of botocore.  If you need a more recent version of botocore please open an issue to that effect.  I'm updating it right now, new release should be out.  please re-open if I'm missing something.  This is similar to how boto3 + awscli pinning works

**Comment by bbakersmith** [comment]
This is definitely a deal-breaker, we're also dropping s3fs for this reason.

**Comment by evinism** [comment]
this should be considered a s3fs issue imo

**Comment by thehesiod** [comment]
TLDR for people at the end of this thread, if you use aiobotocore, and also need awscli and/or boto3, pull those in via the extras like so:
```bash
pip3 install aiobotocore[awscli,boto3]
```

boto3 + awscli have botocore as a hard requirement, we expose these as extras and not hard requirements as you may not need to use them along with aiobotocore.
